web/js/codeworld_shared.js

```diff
@@ -225,31 +250,56 @@ function registerStandardHints(successFunc)
         }
 
         var found = [];
-
+        var hints = Object.keys(codeWorldDocs)
         for (var i = 0; i < hints.length; i++) {
             var hint = hints[i];
-            if (hint.text.startsWith(term)) {
+            if (hint.startsWith(term)){
                 found.push(hint);
             }
         }
 
-        var lines = cm.getValue().split("\n");
-        for (var i=0; i < lines.length; i++) {
-            if (/^\S*\s*::[^:]*$/.test(lines[i])) {
-                var candidate = lines[i].split(" ::")[0];
-                if (candidate.startsWith(term)) {
-                    found.push(createHint(lines[i], 0, candidate.length));
-                }
+        found.sort(function(a, b) {
+            function startsWithLetter(c) {
+                return /^[a-zA-Z].*/.test(c);
             }
-        }
+                if (startsWithLetter(a) && !startsWithLetter(b)) return -1;
+                else if (startsWithLetter(b) && !startsWithLetter(a)) return 1;
+                else return a.toLowerCase() < b.toLowerCase() ? -1 : 1
+            });
 
-        sortHints(found);
+        if (found.length > 0) {
+            var data = {
+                list: found,
+                from: from,
+                to: cur
+            };
 
-        if (found.length > 0) return {
-            list: found,
-            from: from,
-            to: cur
-        };
+            function deleteOldHintDocs(){
+                $(".hint-description").remove()
+            }
+
+            CodeMirror.on(data, 'close', deleteOldHintDocs);
+            CodeMirror.on(data, 'pick', deleteOldHintDocs);
+
+            // Tracking of hint selection
+            CodeMirror.on(
+                data, 'select',
+                function (selection, elem) {
+                    var codeWordInfo = codeWorldDocs[selection],
+                        hihtsWidget = elem.parentElement,
```

Comment:
Misspelling.  It looks like this variable isn't used anyway.

Reply:
Thx, deleted.